From bd49ce86ec47f8e95af96befd4e0aedf47cdb8c3 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 24 Apr 2015 11:54:07 +0200
Subject: [PATCH] Defer modprobe of HW modules, until udev is running

Some HW modules need to load firmware via udev.

https://bugzilla.redhat.com/show_bug.cgi?id=1213077

(cherry picked from commit 8fa11b1e58a60d645c065ac20c02263f6c873d62)
---
 modules.d/95fcoe/parse-fcoe.sh       | 2 +-
 modules.d/95iscsi/parse-iscsiroot.sh | 7 +------
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/modules.d/95fcoe/parse-fcoe.sh b/modules.d/95fcoe/parse-fcoe.sh
index 2fb9b25..d1f6cfd 100755
--- a/modules.d/95fcoe/parse-fcoe.sh
+++ b/modules.d/95fcoe/parse-fcoe.sh
@@ -20,7 +20,7 @@
 # BRCM: Later, should check whether bnx2x is loaded first before loading bnx2fc so do not load bnx2fc when there are no Broadcom adapters
 [ -e /sys/module/fcoe/parameters/create ] || modprobe -a fcoe || die "FCoE requested but kernel/initrd does not support FCoE"
 
-modprobe bnx2fc >/dev/null 2>&1
+initqueue --onetime modprobe -q -b bnx2fc
 
 parse_fcoe_opts() {
     local IFS=:
diff --git a/modules.d/95iscsi/parse-iscsiroot.sh b/modules.d/95iscsi/parse-iscsiroot.sh
index 2bbf3b9..1f6f982 100755
--- a/modules.d/95iscsi/parse-iscsiroot.sh
+++ b/modules.d/95iscsi/parse-iscsiroot.sh
@@ -53,12 +53,7 @@ if [ -n "$iscsiroot" ] ; then
     [ -z "$netroot" ] || [ "$netroot" = "iscsi" ] && netroot=iscsi:$iscsiroot
 fi
 
-modprobe -q -b qla4xxx
-modprobe -q -b cxgb3i
-modprobe -q -b cxgb4i
-modprobe -q -b bnx2i
-modprobe -q -b be2iscsi
-modprobe -q -b crc32c
+initqueue --onetime modprobe --all -q -b qla4xxx cxgb3i cxgb4i bnx2i be2iscsi crc32c
 
 # iscsi_firmware does not need argument checking
 if [ -n "$iscsi_firmware" ] ; then
